src/algebra/algebra/restrict_scalars.lean

```diff
@@ -96,17 +98,34 @@ The preferred way of setting this up is `[module R M] [module S M] [is_scalar_to
 instance : module R (restrict_scalars R S M) :=
 module.comp_hom M (algebra_map R S)
 
+instance : is_scalar_tower R S (restrict_scalars R S M) :=
+⟨λ r S M, by { rw [algebra.smul_def, mul_smul], refl }⟩
+
+/--
+The `R`-algebra homomorphism from the original coefficient algebra `S` to endomorphisms
+of `restrict_scalars R S M`.
+-/
+def restrict_scalars.lsmul : S →ₐ[R] (module.End R (restrict_scalars R S M)) :=
+algebra.lsmul R (restrict_scalars R S M)
+
+end
+
+variables [add_comm_monoid M]
+
+/-- `restrict_scalars.add_equiv` is the additive equivalence with the original module. -/
+@[simps] def restrict_scalars.add_equiv : restrict_scalars R S M ≃+ M :=
```

Comment:
Is this a semilinear map wrt `algebra_map R S`? That would remove the need for `restrict_scalars.add_equiv_map_smul` below.

Reply:
I *think* it is, but to be honest I'd prefer not to to overuse semilinear maps. When we have something that a mathematician would recognise as semilinear (i.e. w.r.t. a star structure), that's great (or when semilinearity saves us significant duplication), but I would only be confused reading this file if the lemma about interacting with scalar multiplication in `S` was hidden in a "semilinearity" statement.